libs/fluid-elements/checkbox/src/lib/checkbox.ts

```diff
@@ -0,0 +1,293 @@
+    // set by the litElement reactivity. In the updated lifeCycle
+    // we need to manually update the ariaChecked attribute here.
+    if (props.has('checked') || props.has('indeterminate')) {
+      // TODO: extract 'mixed' into a shared constant.
+      this.ariaChecked = this.indeterminate ? 'mixed' : this.checked.toString();
+    }
+    // Changing the aria-checked or any observed property in the update, will
+    // add it to the updated properties. When calling super first in, the change
+    // of properties in the update call will trigger an update, as the properties
+    // will have changed after the super.update() call. To prevent an additional
+    // cycle, we make the modifications before calling the super lifecycle
+    super.update(props);
+  }
+
+  /**
+   * Render function of the custom element. It is called when one of the
+   * observedProperties (annotated with @property) changes.
+   */
+  render(): TemplateResult {
+    const classMapData = {
+      'fluid-svg-checkbox': true,
+      'fluid-state--checked': this.checked && !this.indeterminate,
+      'fluid-state--indeterminate': this.indeterminate,
+    };
+
+    return html`
+      <div class="fluid-checkbox">
+        <input
+          type="checkbox"
+          class="fluid-checkbox-input"
+          id="${this._unique}"
+          ?checked="${this.checked}"
+          ?disabled="${this.disabled}"
+          @change="${this._handleInputChange}"
+        />
+        <label for="${this._unique}">
+          <svg
+            class=${classMap(classMapData)}
+            viewBox="0 0 24 24"
+            tabindex="${this.disabled ? -1 : 0}"
+            @keyup="${this._handleKeyup}"
+          >
+            <g transform="translate(12, 12)">
+              <rect
+                class="fluid-svg-checkbox-background"
+                x="-12"
+                y="-12"
+                width="24"
+                height="24"
+              />
+            </g>
+            <rect
+              class="fluid-svg-checkbox-rect"
+              x="4.5"
+              y="4.5"
+              width="15"
+              height="15"
+              stroke-width="1"
+            />
+            <path
+              class="fluid-svg-checkbox-tick"
+              transform="translate(6.3, 7.4), scale(0.77)"
+              d="M4.65833 8.81667L1.18333 5.34167L0 6.51667L4.65833 11.175L14.6583 1.175L13.4833 0L4.65833 8.81667Z"
+            />
+            <rect
+              class="fluid-svg-checkbox-indeterminate"
+              x="7"
+              y="11"
+              width="10"
+              height="1.5"
+              stroke-width="0"
+            />
+          </svg>
+          <span class="fluid-label">
+            <slot></slot>
+          </span>
+        </label>
+      </div>
+    `;
+  }
+
+  /**
+   * Dispatches a change event for the checkbox
+   */
+  private _dispatchChangeEvent(): void {
+    requestAnimationFrame(() => {
```

Comment:
I am interested why you request an animationFrame here? :)

Reply:
Eventually, when grouped elements need to react to a downstream change it seemed like a good idea to defer this to the next cycle.

Weightless does this, but i checked with lion and they are not. 
https://github.com/andreasbm/weightless/blob/6c7965981752e85206d9da459110b2dddeecf9ec/src/lib/behavior/switch/switch-behavior.ts#L119-L124

I'm not sure now and we can remove it if you want.